docs/slicing.rst

```diff
@@ -0,0 +1,360 @@
+As a consequence, wavelength is no longer part of the WCS transformations.
+However, the wavelength value at the location where the `~ndcube.NDCube` was sliced can still be accessed via `~ndcube.NDCube.global_coords`.
+
+.. code-block:: python
+
+  >>> my_2d_cube.global_coords['em.wl']  # doctest: +SKIP
+
+This is true for all coordinates, in both the ``.wcs`` and ``.extra_coords`` objects, that no longer correspond to any array axes after slicing.
+See the :ref:`global_coords` section for more.
+
+.. _ndcube_crop:
+
+Cropping with Real World Coordinates
+------------------------------------
+In addition to slicing by index, `~ndcube.NDCube` supports slicing by real world coordinates via the `~ndcube.NDCube.crop` method.
+This takes two iterables of high level coordinate objects, e.g. `~astropy.time.Time`, `~astropy.coordinates.SkyCoord`, `~astropy.coordinates.SpectralCoord`, `~astropy,units.Quantity` etc.
+Each iterable describes a single location in the data array in real world coordinates.
+The first iterable describes the lower corner of the region of interest and thus contains the lower limit of each real world coordinate.
+The second iterable represents the upper corner of the region of interest and thus contains the upper limit of each real world coordinate.
+The crop method indentifies the smallest rectangular region in the data array that contains both the lower and upper limits in all the real world coordinates, and crops the `~ndcube.NDCube` to that region.
+It does not rebin or interpolate the data.  The order of the high level coordinate objects in each iterable must be the same as that expected by `astropy.wcs.WCS.world_to_array_index`, namely in world order.
+
+.. code-block:: python
+
+  >>> import astropy.units as u
+  >>> from astropy.coordinates import SkyCoord, SpectralCoord
+  >>> from sunpy.coordinates.frames import Helioprojective
+  >>> # Use coordinate objects to mark the lower limit of the region of interest.
+  >>> lower_corner = [SpectralCoord(1.04e-9, unit=u.m),
+  ...                 SkyCoord(Tx=1, Ty=0.5, unit=u.deg, frame=Helioprojective)]
+  >>> upper_corner = [SpectralCoord(1.08e-9, unit=u.m),
+  ...                 SkyCoord(Tx=1.5, Ty=1.5, unit=u.deg, frame=Helioprojective)]
+  >>> my_cube_roi = my_cube.crop(lower_corner, upper_corner)
+
+.. _sequence_slicing:
+
+Slicing NDCubeSequences
+=======================
+As with `~ndcube.NDCube`, `~ndcube.NDCubeSequence` is sliced by applying the standard slicing API.
+The `~ndcube.NDCubeSequence` slicing infrastrcuture determines which cubes should be kept based on the sequence axis input, then passes the rest of the slicing off to desired NDCubes.
+Thus the data arrays, WCS transformations, masks, uncertainty arrays, and extra coordinates are all altered accorindingly in each relevant sub-cube.
+Say we have three NDCubes in an `~ndcube.NDCubeSequence`, each of shape ``(4, 4, 5)``.
+
+.. expanding-code-block:: python
```

Comment:
Would it not be better if this were moved to just above the code block where it is used?